core/src/core/transaction.rs

```diff
@@ -511,52 +511,44 @@ impl TransactionBody {
 			// If we are verifying sort order then verify and
 			// return an error if not sorted lexicographically.
 			body.verify_sorted()?;
-			Ok(body)
 		} else {
 			// If we are not verifying sort order then sort in place and return.
-			let mut body = body;
 			body.sort();
-			Ok(body)
 		}
+		Ok(body)
 	}
 
 	/// Builds a new body with the provided inputs added. Existing
 	/// inputs, if any, are kept intact.
 	/// Sort order is maintained.
-	pub fn with_input(self, input: Input) -> TransactionBody {
-		let mut new_ins = self.inputs;
-		new_ins.push(input);
-		new_ins.sort();
-		TransactionBody {
-			inputs: new_ins,
-			..self
-		}
+	pub fn with_input(mut self, input: Input) -> TransactionBody {
```

Comment:
These are actually (potentially) breaking changes.
The original intent was for these to be factories that left the original body unchanged, creating new instances each time.
I'm not sure if we actually _needed_ this guarantee though (to leave the original one unchanged).

:+1: If you are confident this is ok to change.

Reply:
Hmmm -- I don't think that's quite true -- the contract of these functions is a pass by ownership (I'm only touching the mutability).

Because the self passed in is an owned self, the passed in instance is consumed so I figured it's OK to trash it.

Reply:
No you're right - we're consuming the instance here each time. :+1: